### PR TITLE
Change naming of bolus absorption and add verb get_ to number_of

### PR DESCRIPTION
**Pull request by JohanWBorg**
In response to the final comments of #2032.
- Change name of bolus absorption to instantaneous absorption
- Add verb _get to number_of_peripheral/transit_compartments
- Added has_ absorption type and number_of functions to /modeling __innit__ file

### CODE REVIEW

File: src/pharmpy/modeling/odes.py [hunk 1]
```diff
@@ -1505,7 +1505,7 @@ def has_first_order_absorption(model: Model):
 
     """
 
-    odes = model.statements.ode_system
+    odes = get_and_check_odes(model)
     if odes is None:
```
Comment (rikardn) [inline src/pharmpy/modeling/odes.py:1509]: Now this is not needed. `get_and_check_odes` guarantees the return of a `CompartmentalSystem` (else it raises)

**Review by rikardn — CHANGES_REQUESTED** [review]
One comment

File: src/pharmpy/modeling/odes.py (hunk 1 above)
Comment (JohanWBorg) [reply src/pharmpy/modeling/odes.py:1509]: Oh sorry, I'll fix it right away!

**Comment by rikardn** [comment]
Super!

**Pull request merged**